dev/io.openliberty.data.internal.persistence/src/io/openliberty/data/internal/persistence/service/RecordTransformer.java

```diff
@@ -0,0 +1,309 @@
+
+            if (genericSignature == null) {
+                this.setterSignature = this.fieldSignature = this.getterSignature = genericSignature;
+            } else {
+                this.fieldSignature = genericSignature;
+                this.setterSignature = "(" + fieldSignature + ")V";
+                this.getterSignature = "()" + fieldSignature;
+            }
+        }
+
+        public String getMethodName(String prefix) {
+            return prefix + name.substring(0, 1).toUpperCase() + name.substring(1);
+        }
+
+        @Override
+        public String toString() {
+            return "RecordComponentInfo [name=" + name + ", type=" + type + ", descriptor=" + descriptor + ", getterDescriptor=" + getterDescriptor + ", setterDescriptor="
+                   + setterDescriptor + ", fieldSignature=" + fieldSignature + ", setterSignature=" + setterSignature + ", getterSignature=" + getterSignature + "]";
+        }
+
+    }
+
+    /**
+     * Generates an entity class from a record class.
+     *
+     * The generated entity class will have a public field, getter method, and setter method
+     * for each record component.
+     *
+     * The generated entity class will have a public no-args constructor, a public constructor
+     * that accepts the corresponding record, and a public toRecord() method that returns a
+     * new instance of the record.
+     *
+     * Initially copied from @nmittles pull #25248
+     *
+     * @param recordClass     - the record class that will be transformed into an entity class
+     * @param entityClassName - the name of the entity class
+     *
+     * @return the byte array representation of the generated entity class
+     */
+    public static byte[] generateEntityClassBytes(Class<?> recordClass, String entityClassName) {
+        final boolean trace = TraceComponent.isAnyTracingEnabled();
+
+        final String CTOR = "<init>";
+        final String TO_RECORD = "toRecord";
+
+        // Names of classes used in generating an entity class from a record
+        final String _recordClassName = recordClass.getName().replace(".", "/");
+        final String _recordClassNameDesc = org.objectweb.asm.Type.getDescriptor(recordClass);
+
+        final String _entityClassName = entityClassName.replace('.', '/');
+        final String _objectClassName = Object.class.getName().replace('.', '/');
+
+        final String _sourceFileName = entityClassName.substring(entityClassName.lastIndexOf(".") + 1) + ".java";
+
+        if (recordClass.getTypeParameters().length != 0) {
+            throw new IllegalStateException("The " + recordClass.getName() + " record has one or more generic types " +
+                                            Arrays.asList(recordClass.getTypeParameters()) + ". This record class cannot be transformed into "
+                                            + " and entity class since entity classes cannot be generic."); //TODO NLS
```

Comment:
IllegalStateException Javadoc describes it as, "Signals that a method has been invoked at an illegal or inappropriate time. In other words, the Java environment or Java application is not in an appropriate state for the requested operation."  I'm not quite sure what exception to use instead though.  Maybe the generic DataException?

```suggestion
            throw new DataException("The " + recordClass.getName() + " record has one or more generic types " +
                                            Arrays.asList(recordClass.getTypeParameters()) + ". This record class cannot be transformed into "
                                            + " an entity class since entity classes cannot be generic."); //TODO NLS
```